wallet/test/Test/Pos/Wallet/Web/Mode.hs

```diff
@@ -0,0 +1,444 @@
+  where
+    dbGetBlock = DB.dbGetBlockPureDefault
+    dbGetUndo = DB.dbGetUndoPureDefault
+    dbGetHeader = DB.dbGetHeaderPureDefault
+
+instance HasConfiguration => MonadBlockDBGeneric (Some IsHeader) SscBlock () WalletTestMode
+  where
+    dbGetBlock = DB.dbGetBlockSscPureDefault
+    dbGetUndo = DB.dbGetUndoSscPureDefault
+    dbGetHeader = DB.dbGetHeaderSscPureDefault
+
+instance
+    HasConfiguration =>
+    MonadBlockDBGenericWrite BlockHeader Block Undo WalletTestMode
+  where
+    dbPutBlund = DB.dbPutBlundPureDefault
+
+instance HasConfiguration => MonadGState WalletTestMode where
+    gsAdoptedBVData = gsAdoptedBVDataDefault
+
+instance MonadFormatPeers WalletTestMode where
+    formatKnownPeers _ = pure Nothing
+
+instance MonadKnownPeers WalletTestMode where
+    updatePeersBucket _ _ = pure True
+
+----------------------------------------------------------------------------
+-- Wallet instances
+----------------------------------------------------------------------------
+
+instance HasLens WalletState WalletTestContext WalletState where
+    lensOf = wtcWalletState_L
+
+-- For MonadUpdates
+instance HasShutdownContext WalletTestContext where
+    shutdownContext = wtcShutdownContext_L
+
+-- For MonadRecoveryInfo
+instance HasLens RecoveryHeaderTag WalletTestContext RecoveryHeader where
+    lensOf = wtcRecoveryHeader_L
+
+-- For BListener
+instance HasLens StateLock WalletTestContext StateLock where
+    lensOf = wtcStateLock_L
+
+-- For MonadBlockchainInfo
+instance HasLens LastKnownHeaderTag WalletTestContext LastKnownHeader where
+    lensOf = wtcLastKnownHeader_L
+
+instance HasLens ProgressHeaderTag WalletTestContext ProgressHeader where
+    lensOf = wtcProgressHeader_L
+
+instance HasLens ConnectedPeers WalletTestContext ConnectedPeers where
+    lensOf = wtcConnectedPeers_L
+
+-- For reporting
+instance HasNodeType WalletTestContext where
+    getNodeType _ = NodeCore -- doesn't really matter, it's for reporting
+
+-- TODO may be used for callback on tx processing in future.
+instance HasLens StateLockMetrics WalletTestContext StateLockMetrics where
+    lensOf = lens (const emptyStateMetrics) const
+      where
+        emptyStateMetrics = StateLockMetrics
+            { slmWait = const $ pure ()
+            , slmAcquire = const $ pure ()
+            , slmRelease = const $ pure ()
+            }
+
+-- TODO remove HasCompileInfo here
+-- when getNewAddressWebWallet won't require MonadWalletWebMode
+instance (HasCompileInfo, HasConfigurations) => MonadAddresses WalletTestMode where
+    type AddrData WalletTestMode = (AccountId, PassPhrase)
+    getNewAddress = getNewAddressWebWallet
+    getFakeChangeAddress = pure largestHDAddressBoot
+
+instance MonadKeys WalletTestMode where
+    getSecret = view wtcUserSecret_L >>= atomically . STM.readTVar
+    modifySecret f = do
+        us <- view wtcUserSecret_L
+        void $ atomically $ STM.modifyTVar us f
```

Comment:
`void` is extra here

Reply:
cheto ya somnevaus'

Reply:
Really, `modifyTVar` is strange in this sense. That's why we need `modifyTVarS` sometimes